src/modules/Update.rb

```diff
@@ -773,22 +773,32 @@ def installed_product
     end
 
     BACKUP_DIR = "var/adm/backup/system-upgrade"
-    # Creates backup with name based on `name` contaings everything
-    # matching globs in `paths`.
-    # @param name[String] name for backup file. Use a number prefix to run
-    #   the restore scripts in the expected order. Use a bash friendly name ;)
+    # Creates the backup with name based on `name` containing everything matching globs in `paths`
+    #
     # @note Can be called only after target root is mounted.
     #
+    # @param name [String] name for backup file. Use a number prefix to run the restore scripts in
+    #   the expected order. Use a bash friendly name ;)
+    # @param paths [Array<String>] paths to files that must be included
+    #
     # @example to store repos file and credentials directory
-    #   Update.create_backup("0100-repos", ["/etc/zypp/repos.d/*", "/etc/zypp/credentials"])
+    #   Yast::Update.create_backup("0100-repos", ["/etc/zypp/repos.d/*", "/etc/zypp/credentials"])
     def create_backup(name, paths)
-      log.info "Creating tarball for #{name} including #{paths}"
       mounted_root = Installation.destdir
 
+      # Copy the os-release file (bsc#1097297)
+      log.info "Copying the os-release file}"
+      ::FileUtils.cp(
```

Comment:
How does it behave when the `/etc/os-release` file is missing? That should normally not happen, but users sometimes do really weird things...

Reply:
I was missing a file removal but it turned out that the `Update.clean_backup` method removes the directory repository recursively so everything is removed. But maybe this should be added as a comment here...

Reply:
You're right, the `#clean_backup` will remove it too.

Ok, I'll add a comment clarifying this.